src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -132,6 +143,54 @@ public static class PrereleaseCompatibilityTransformer
             return transformedSarifLog;
         }
 
+        private static bool ApplyChangesFromTC34(JObject sarifLog)
+        {
+            // TODO: update sarif version
+
+            // https://github.com/oasis-tcs/sarif-spec/issues/361
+            ConvertAllStateStringsToMultiFormatMessageStrings(sarifLog);
+            return true;
+        }
+
+        private static void ConvertAllStateStringsToMultiFormatMessageStrings(JObject run)
+        {
+            string[] statePathsToUpdate =
+            {
+                "inlineExternalProperties[].results[].graphTraversals[].edgeTraversals[].finalState",
```

Comment:
>inlineExternalProperties [](start = 17, length = 24)

replace this with a new helper that accepts a hashset of strings, each of which is a property name like finalState. then make a single pass over the JSON, and call your callback with any property value where the property name is included in the set.

Reply:
tracking item #214

---
In reply to: [273270880](https://github.com/Microsoft/sarif-sdk/pull/1412#discussion_r273270880) [](ancestors = 273270880)